### PR TITLE
support choose gitUrlPattern

### PR DESCRIPTION
**Pull request by superyyrrzz**
#677 
@chenkennt @hellosnow @vwxyzh @qinezh @ansyral

### CODE REVIEW

File: src/docfx.website.themes/common/common.js [hunk 1]
```diff
@@ -24,83 +24,126 @@ function getDirectoryName(path) {
...
+        repo = repo.substr(0, repo.length - 4);
+    }
+
+    var patternName = getPatternName(repo, gitUrlPattern);
+    if (!patternName) return '';
+    return gitUrlPatternItems[patternName].generateUrl(repo, branch, remote.path, startLine);
+}
+
+function getPatternName(repo, gitUrlPattern)
+{
```
Comment (hellosnow) [inline src/docfx.website.themes/common/common.js]: move the bracket to line above?

File: src/docfx.website.themes/common/common.js [hunk 2]
```diff
@@ -24,83 +24,126 @@ function getDirectoryName(path) {
...
+    var patternName = '';
+    if (gitUrlPattern && gitUrlPattern.toLowerCase() in gitUrlPatternItems) {
+        patternName = gitUrlPattern.toLowerCase();
+    } else {
+        for (var p in gitUrlPatternItems) {
+            if (gitUrlPatternItems[p].testRegex.test(repo)) {
+                patternName = p;
```
Comment (hellosnow) [inline src/docfx.website.themes/common/common.js]: >patternName = p; [](start = 16, length = 16)

if muptiple testRegex is matched, will get the last one? How about return directly if matched?

Comment (superyyrrzz) [inline src/docfx.website.themes/common/common.js]: sounds nice

---
In reply to: [78559120](https://github.com/dotnet/docfx/pull/696#discussion_r78559120) [](ancestors = 78559120)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**